### PR TITLE
[Core]Adding flag possibilty in the fast transfer between model parts process

### PR DESCRIPTION
**Pull request by loumalouomega**
- Adding flag possibilty
- Moving to a cpp
- Adding CPP test

### CODE REVIEW

**Comment by loumalouomega** [comment]
@RiccardoRossi I change the thing to use real flags

File: kratos/processes/fast_transfer_between_model_parts_process.cpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        }
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+EntityTransfered FastTransferBetweenModelPartsProcess::ConvertEntity(const std::string& Str)
```
Comment (RiccardoRossi) [inline kratos/processes/fast_transfer_between_model_parts_process.cpp]: then, do you still need this funciton? (after using "real flags")

**Comment by loumalouomega** [comment]
Done, now using just enums. @RiccardoRossi I moved the declaration in add_process_to_python because for boost::python for enum class is necessary to declare a context that breaks the declaration of the other processes. @roigcarlo says that with pybind this does not happen

**Comment by loumalouomega** [comment]
@RiccardoRossi ping

**Comment by loumalouomega** [comment]
Remember remember

**Comment by loumalouomega** [comment]
Ping @RiccardoRossi

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**